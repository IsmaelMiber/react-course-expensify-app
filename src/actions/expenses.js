import uuid from 'uuid';

//ADD_EXPENSE
export const addExpense = ({description = '', note = '', amount = 0, createAt = 0} = {}) => (
    {
        type: "ADD_EXPENSE",
        expense: {
            id: uuid(),
            amount,
            description,
            note,
            createAt,
        }
    }
);

//REMOVE_EXPENSE
export const removeExpense = (idOfTargetExpense) => (
    {
        type: "REMOVE_EXPENSE",
        idOfTargetExpense,
    }
);

//EDIT_EXPENSE
export const editExpense = (id, updates) => (
    {
        type: "EDIT_EXPENSE",
        id,
        updates,
    }
);