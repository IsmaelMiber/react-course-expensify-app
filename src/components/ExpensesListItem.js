import React from 'react';
import { Link } from 'react-router-dom';

const ExpensesListItem = (props) => (
    <div>
        <p><Link to={`/edit/${props.id}`}>description: {props.description}</Link></p>
        <p>amount: {props.amount}</p>
        <p>createdAt: {props.createAt}</p>
        <hr />
    </div>
);

export default ExpensesListItem;