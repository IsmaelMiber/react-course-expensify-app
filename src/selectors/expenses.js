import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter( ({ createAt, description }) => {
            const createAtMoment = moment(createAt);
            const startDateMatch = startDate ? startDate.isSameOrBefore(createAtMoment, 'day') : true;
            const endDateMatch = endDate ? endDate.isSameOrAfter(createAtMoment, 'day') : true;
            const textMatch = description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort( (a, b) => {
        if(sortBy === 'date') {
            return a.createAt < b.createAt ? 1 : -1;
        } else {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};