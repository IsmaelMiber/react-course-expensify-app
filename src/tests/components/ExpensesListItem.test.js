import React from 'react';
import ExpensesListItem from '../../components/ExpensesListItem';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test("Should render ExpensesListItem", () => {
    const wrapper = shallow(<ExpensesListItem { ...expenses[1] } />);
    expect(wrapper).toMatchSnapshot();
});