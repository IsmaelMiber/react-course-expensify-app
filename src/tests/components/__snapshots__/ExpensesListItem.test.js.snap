// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesListItem 1`] = `
<div>
  <p>
    <Link
      replace={false}
      to="/edit/2"
    >
      description: 
      iphone x
    </Link>
  </p>
  <p>
    amount: 
    10
  </p>
  <p>
    createdAt: 
    345600000
  </p>
  <hr />
</div>
`;
